src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1

```diff
@@ -973,6 +973,262 @@ function Test-ApplicationGatewayCRUDRewriteRuleSetWithConditions
 	}
 }
 
+function Test-ApplicationGatewayCRUDRewriteRuleSetWithUrlConfiguration
```

Comment:
Can you test your changes in existing test function like `ApplicationGatewayCRUDRewriteRuleSet` or `ApplicationGatewayCRUDSubItems`? I'm concerned that there are a lot of very similar tests in AppGw when each of them takes about 30 minutes to run

Reply:
For v2 gateways, any update should take maximum 5 mins. In this test, there are two create/update hence it will take maximum 10-12 minutes. You must be seeing 30 mins in the test cases when there are many creates/updates. We have around 20 mins time for V1 gateways so you might see high test running time for V1 gateways.

Is there any check in gate where these tests run in record mode? I won't mind adding each tests with different scenario if they are running locally only and check in gates run the playback tests.

I agree that there are many tests which can be merged into single test like RewriteRuleCRUD and RewriteRuleCRUDwithConditions. I think it would be better to open a work item on AppGW team to merge them and it should not be the part of this PR.

Apart from that, it would require making changes in the existing test cases which would require loop from the tests author itself to avoid any potential assert missing.